thrift-logger/src/main/java/com/pinterest/singer/client/AuditableLogbackThriftLogger.java

```diff
@@ -132,23 +160,35 @@ private void appendInternal(byte[] partitionKey, byte[] message, long timeNanos,
     try {
```

Comment:
How about have the following change to minimize boilerplate code?

diff --git a/thrift-logger/src/main/java/com/pinterest/singer/client/AuditableLogbackThriftLogger.java b/thrift-logger/src/main/java/com/pinterest/singer/client/AuditableLogbackThriftLogger.java
index 5a8b0dc..13a9302 100644
--- a/thrift-logger/src/main/java/com/pinterest/singer/client/AuditableLogbackThriftLogger.java
+++ b/thrift-logger/src/main/java/com/pinterest/singer/client/AuditableLogbackThriftLogger.java
@@ -131,7 +131,7 @@ public class AuditableLogbackThriftLogger extends LogbackThriftLogger {
     if (enableLoggingAudit) {
       headers = auditHeadersGenerator.generateHeaders();
     }
-    appendInternal(partitionKey, message,timeNanos, headers);
+    append(partitionKey, message,timeNanos, headers);
   }
 
   @Override
@@ -147,7 +147,7 @@ public class AuditableLogbackThriftLogger extends LogbackThriftLogger {
         }
       }
       byte[] messageBytes = ThriftCodec.getInstance().serialize(thriftMessage);
-      appendInternal(partitionKey, messageBytes, timeNanos, headers);
+      append(partitionKey, messageBytes, timeNanos, headers);
     } catch (TException e) {
       OpenTsdbMetricConverter.incr(THRIFT_LOGGER_ERROR_TEXCEPTION,
           "topic=" + topic, "host=" + HOST_NAME);
@@ -155,36 +155,23 @@ public class AuditableLogbackThriftLogger extends LogbackThriftLogger {
     }
   }
 
-  private void appendInternal(byte[] partitionKey, byte[] message, long timeNanos,
+  private void append(byte[] partitionKey, byte[] message, long timeNanos,
                               LoggingAuditHeaders headers) throws LogbackException {
-    try {
       LogMessage logMessage = new LogMessage()
-          .setTimestampInNanos(timeNanos)
-          .setMessage(message);
+          .setTimestampInNanos(timeNanos).setMessage(message);
 
       if(this.enableLoggingAudit && headers != null) {
         logMessage.setLoggingAuditHeaders(headers);
         OpenTsdbMetricConverter.incr(AUDIT_THRIFT_LOGGER_HEADERS_ADDED_TO_LOG_MESSAGE_COUNT,
             "topic=" + topic, "host=" + HOST_NAME);
       }
-
-      if (partitionKey != null) {
-        logMessage.setKey(partitionKey);
-      }
-      appender.doAppend(logMessage);
-      OpenTsdbMetricConverter.incr(THRIFT_LOGGER_COUNT_METRIC, "topic=" + topic, "host=" + HOST_NAME);
-
+      super.append(logMessage, partitionKey);
       if (this.enableLoggingAudit && headers != null && AuditableLogbackThriftLoggerFactory.getLoggingAuditClient() != null) {
           AuditableLogbackThriftLoggerFactory.getLoggingAuditClient().audit(this.topic, headers, true,
               System.currentTimeMillis());
           OpenTsdbMetricConverter.incr(AUDIT_THRIFT_LOGGER_AUDITED_MESSAGE_COUNT,
               "topic=" + topic, "host=" + HOST_NAME);
-
       }
-    } catch (LogbackException e) {
-      OpenTsdbMetricConverter.incr(THRIFT_LOGGER_ERROR_LOGBACKEXCEPTION, "topic=" + topic, "host=" + HOST_NAME);
-      throw e;
-    }
   }
 
   public boolean shouldAudit(){
diff --git a/thrift-logger/src/main/java/com/pinterest/singer/client/LogbackThriftLogger.java b/thrift-logger/src/main/java/com/pinterest/singer/client/LogbackThriftLogger.java
index 1934f4f..5c5fa98 100644
--- a/thrift-logger/src/main/java/com/pinterest/singer/client/LogbackThriftLogger.java
+++ b/thrift-logger/src/main/java/com/pinterest/singer/client/LogbackThriftLogger.java
@@ -70,13 +70,13 @@ public class LogbackThriftLogger extends BaseThriftLogger {
     this.appender = appender;
   }
 
-  @Override
-  public void append(byte[] partitionKey, byte[] message, long timeNanos) {
+  /**
+   * This method is also called in AuditableLogackThriftLogger to minimize boilerplate code.
+   * @param logMessage  logMessage
+   * @param partitionKey partition key
+   */
+  protected void append(LogMessage logMessage, byte[] partitionKey) {
     try {
-      LogMessage logMessage = new LogMessage()
-          .setTimestampInNanos(timeNanos)
-          .setMessage(message);
-
       if (partitionKey != null) {
         logMessage.setKey(partitionKey);
       }
@@ -90,6 +90,12 @@ public class LogbackThriftLogger extends BaseThriftLogger {
     }
   }
 
+  @Override
+  public void append(byte[] partitionKey, byte[] message, long timeNanos) {
+      LogMessage logMessage = new LogMessage().setTimestampInNanos(timeNanos).setMessage(message);
+      append(logMessage, partitionKey);
+  }
+
   @Override
   public void append(byte[] partitionKey, TBase thriftMessage, long timeNanos) throws TException {
     try {

Reply:
I applied your diff and removed boilerplate code. thanks for this great suggestion.